src/mmds.jl

```diff
@@ -0,0 +1,208 @@
+"""
+(Non)Metric Multidimensional Scaling
+
+Use this class to perform (non)metric multidimensional scaling.
+
+There are two calling options, specified via the required keyword argument `distances`:
+
+```julia
+mds = fit(MDS, X; distances=false, maxoutdim=size(X,1)-1)
+```
+
+where `X` is the data matrix. Distances between pairs of columns of `X` are computed using the Euclidean norm.
+This is equivalent to performing PCA on `X`.
+
+```julia
+mds = fit(MDS, D; distances=true, maxoutdim=size(D,1)-1)
+```
+
+where `D` is a symmetric matrix `D` of distances between points.
+
+In addition, the `metric` parameter specifies type of MDS, By default, it is assigned with `nothing` value
+which results in performing *metric MDS* with dissimilarities calculated as Euclidian distances.
+
+An arbitrary transformation function can be provided to `metric` parameter to
+perform metric MDS with transformed proximities. The function has to accept two parameters,
+a vector of proximities and a vector of distances, in order to calculate disparities
```

Comment:
What do the vectors represent? I.e., if `p` is a proximity vector, what is the meaning of `p[i]`? Likewise, what is the meaning of `d[i]`?

Reply:
Proximity and distance vectors are compact (1D) representations of upper triangular part of proximity and distance matrices correspondingly.